internal/helm/templateloader/loader.go

```diff
@@ -429,6 +450,106 @@ func transformSize(value interface{}) (interface{}, error) {
 	return nil, errors.Errorf("invalid type for size %T", value)
 }
 
+func transformStringTemplate(value interface{}) (interface{}, error) {
+	return templatetypes.StringTemplate{Value: fmt.Sprintf("%v", value)}, nil
+}
+
+func transformStringTemplateList(data interface{}) (interface{}, error) {
+	switch value := data.(type) {
+	case string:
+		return templatetypes.StringTemplateList{templatetypes.StringTemplate{Value: value}}, nil
+	case []interface{}:
+		res := templatetypes.StringTemplateList{}
+		for _, v := range value {
+			res = append(res, templatetypes.StringTemplate{ Value: fmt.Sprintf("%v", v)})
+		}
+		return res, nil
+	default:
+		return data, errors.Errorf("invalid type %T for string list", value)
+	}
+}
+
+func transformShellCommandTemplate(value interface{}) (interface{}, error) {
+	if str, ok := value.(string); ok {
+		return shellwords.Parse(str)
+	}
+	return value, nil
+}
+
+func transformHostsListTemplate(data interface{}) (interface{}, error) {
+	return transformListOrMapping(data, ":", false), nil
+}
+
+func toStringList(value map[string]interface{}, separator string, allowNil bool) []string {
+	output := []string{}
+	for key, value := range value {
+		if value == nil && !allowNil {
+			continue
+		}
+		output = append(output, fmt.Sprintf("%s%s%s", key, separator, value))
+	}
+	sort.Strings(output)
+	return output
+}
+
+func toString(value interface{}, allowNil bool) interface{} {
+	switch {
+	case value != nil:
+		return fmt.Sprint(value)
+	case allowNil:
+		return nil
+	default:
+		return ""
+	}
+}
+
+func toMapStringString(value map[string]interface{}, allowNil bool) map[string]interface{} {
+	output := make(map[string]interface{})
+	for key, value := range value {
+		output[key] = toString(value, allowNil)
+	}
+	return output
+}
+
+func transformListOrMapping(listOrMapping interface{}, sep string, allowNil bool) interface{} {
+	switch value := listOrMapping.(type) {
+	case map[string]interface{}:
+		return toStringList(value, sep, allowNil)
+	case []interface{}:
+		return listOrMapping
+	}
+	panic(errors.Errorf("expected a map or a list, got %T: %#v", listOrMapping, listOrMapping))
```

Comment:
KEEP CALM... and don't panic 😄 
Could we return an error instead?

Reply:
This was shamelessly stolen fro cli/cli/compose/types. I raised an eyebrow when I saw that too :) , but the schema validation step makes this impossible to trigger.